docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,425 @@
+* `--cni-bin-dir=/opt/cni/bin` specifies where to look for the actual CNI binaries
+* `--cluster-dns=10.96.0.10` use this cluster-internal DNS server for `nameserver`
+   entries in Pods' `/etc/resolv.conf`
+* `--cluster-domain=cluster.local` uses this cluster-internal DNS domain for
+   `search` entries in Pods' `/etc/resolv.conf`
+* `--client-ca-file=/etc/kubernetes/pki/ca.crt` authenticates requests to the Kubelet
+   API using this CA certificate.
+* `--authorization-mode=Webhook` authorizes requests to the Kubelet API by `POST`-ing
+   a `SubjectAccessReview` to the API Server
+* `--cadvisor-port=0` disables cAdvisor from listening to `0.0.0.0:4194` by default.
+   cAdvisor will still be run inside of the kubelet and its API can be accessed at
+   `https://{node-ip}:10250/stats/`. If you want to enable cAdvisor to listen on a
+   wide-open port, run:
+
+   ```
+   sed -e "/cadvisor-port=0/d" -i /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+   systemctl daemon-reload
+   systemctl restart kubelet
+   ```
+* `--rotate-certificates` auto rotate the kubelet client certificates by requesting new 
+   certificates from the kube-apiserver when the certificate expiration approaches.
+* `--cert-dir`the directory where the TLS certs are located.
+
+### Use kubeadm with other CRI runtimes
+
+Since the [Kubernetes 1.6 release](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#node-components-1),
+Kubernetes has started using CRI as the default container runtime..
+the container runtime used by kubeadm is Docker, which is enabled through the built-in
+`dockershim` in `kubelet` mechanisms.
+
+Other CRI-based runtimes include::
+
+- [cri-o](https://github.com/kubernetes-incubator/cri-o)
+- [frakti](https://github.com/kubernetes/frakti)
+- [rkt](https://github.com/kubernetes-incubator/rktlet)
+
+After you have successfully installed `kubeadm` and `kubelet`, please follow
+these two steps:
+
+1. Install runtime shim on every node. You will need to follow the installation
+   document in the runtime shim project listing above.
+
+1. Configure kubelet to use remote CRI runtime. Please remember to change
+   `RUNTIME_ENDPOINT` to your own value like `/var/run/{your_runtime}.sock`:
+
+```shell
+  $ cat > /etc/systemd/system/kubelet.service.d/20-cri.conf <<EOF
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --container-runtime-endpoint=$RUNTIME_ENDPOINT --feature-gates=AllAlpha=true"
+EOF
+  $ systemctl daemon-reload
+```
+
+Now `kubelet` is ready to use the specified CRI runtime, and you can continue
+with `kubeadm init` and `kubeadm join` workflow to deploy Kubernetes cluster.
+
+### Using internal IPs in your cluster
+
+In order to set up a cluster where the master and worker nodes communicate with internal IP addresses (instead of publicly addressable ones), you will need to follow these steps.
+
+1. When running init, you must make sure you specify an internal IP for the apiserver's bind address, like so:
```

Comment:
Global comment: Use "the API server" for referring to the API server generically.